canu/generate/switch/config/config.py

```diff
@@ -1264,6 +1270,10 @@ def parse_sls_for_config(input_json):
             sls_variables["CMN_NETMASK"] = sls_variables["CMN"].netmask
             sls_variables["CMN_PREFIX_LEN"] = sls_variables["CMN"].prefixlen
             sls_variables["CMN_NETWORK_IP"] = sls_variables["CMN"].ip
+            sls_variables["CMN_ASN"] = sls_network.get("ExtraProperties", {}).get(
```

Comment:
That's pretty slick wrt Python-wrangling.  If the variable gets set to {} what ends up in the config?  And is this a sane default?

Related to the above question... If I have CSM 1.0 SLS data w/o the BGP data, what does this produce a decent config or should we flat out fail if that data isn't there?

Reply:
I have a check for the CMN vlan, but not explicitly for the CMN ASN.
https://github.com/Cray-HPE/canu/blob/2c3745c3e85d84b856d8a92d8b37bf4387c9df71/canu/generate/switch/config/config.py#L495-L511

the CSM 1.0 templates are using the traditional hard coded ASNs since the SLS does not have that data.